### PR TITLE
fix(logger): handle errors

### PR DESCRIPTION
**Pull request by viceice**
* log error details even on sub errors
* add special logging for `HttpError`

### CODE REVIEW

File: lib/logger/utils.ts [hunk 1]
```diff
@@ -49,6 +51,47 @@ const contentFields = [
...
+  // handle got error
+  if (err instanceof RequestError) {
+    const options: Record<string, unknown> = {
+      headers: clone(err.options.headers),
+      url: err.options.url?.toString(),
+    };
+    response.options = options;
+
+    for (const k of ['username', 'password', 'method', 'http2']) {
+      options[k] = err.options[k];
```
Comment (rarkins) [inline lib/logger/utils.ts:78]: Will they still be sanitized correctly?

**Review by rarkins — DISMISSED** [review]
Looks good, assume you tested it :)

File: lib/logger/utils.ts (hunk 1 above)
Comment (viceice) [reply lib/logger/utils.ts:78]: sure, the error simply got patched before sanitizing, maybe we can remove the errSerializer entirely later.

File: lib/logger/utils.ts [hunk 2]
```diff
@@ -49,6 +51,48 @@ const contentFields = [
   'yarnLockParsed',
 ];
 
+export default function prepareError(err: Error): Record<string, unknown> {
+  const response: Record<string, unknown> = {
+    ...err,
+  };
+
+  // Can maybe removed?
```
Comment (rarkins) [inline lib/logger/utils.ts:59]: Should we remove these now? I don't see how this helps anything

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/logger/utils.ts (hunk 2 above)
Comment (viceice) [reply lib/logger/utils.ts:59]: it only helps, if the message or stack are non-enumerable properties on a custom error class. But i think we can remove them anyways. If we see something missin gin logs we can add handling on demand.

Comment (viceice) [reply lib/logger/utils.ts]: OK, looks like this is still needed, because `message` and `stack` are not copied by default. 😢 So i'll revert the changes

**Review by rarkins — DISMISSED** [review]

**Comment by viceice** [comment]
should work as expected now

File: lib/logger/__snapshots__/err-serializer.spec.ts.snap [hunk 3]
```diff
@@ -33,7 +33,23 @@ Array [
...
+      "accept": "application/json",
+      "accept-encoding": "gzip, deflate",
+      "authorization": "Basic OnRva2Vu",
+      "host": "github.com",
+      "user-agent": "https://github.com/renovatebot/renovate",
+    },
+    "method": "POST",
+    "url": "https://:token@github.com/api",
+  },
+]
```
Comment (viceice) [inline lib/logger/__snapshots__/err-serializer.spec.ts.snap:49]: this is the raw request captured from nock, so not sanitized.

**Review by rarkins — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**